lib/generators/plugin/migration_generator.rb

```diff
@@ -0,0 +1,72 @@
+module Plugin
+  class MigrationGenerator < Rails::Generators::Base
+    class_option :plugin_name, :required => true
+    class_option :plugin_source
+
+    attr_accessor :migration_pattern
+
+    hook_for :orm, :required => true do |instance, klass|
+      new_migration = instance.invoke klass, nil, nil, destination_root: instance.send(:migration_destination)
+      # the task return a suggested new migration name. If the migration already
+      # exists, the file with older timestamp would remain. In this case we
+      # should search for the older file with the same name part.
+      instance.migration_pattern = new_migration.first
+    end
+
+    def initialize(*args)
+      super
+      self.destination_root = migration_destination
+    end
+
+    def copy_latest_migration
+      destination_file = File.join(
+        destination_root,
+        'db/migrate',
+        destination_file_name
+      )
+
+      unless options['pretend']
+        # There is no rename task for some reason...
+        copy_file migration_file, destination_file
+        remove_file File.join('db/migrate', migration_file)
+      end
+      [destination_file]
+    end
+
+    private
+
+    def extract_migration_name(migration_file_name)
+      File.basename(migration_file_name).sub(/^\d*_/, '').sub(/\.rb$/, '')
+    end
+
+    def source_paths
+      [File.join(migration_destination, 'db/migrate')]
+    end
+
+    def migration_destination
+      File.join(destination_root, plugin_source)
+    end
+
+    def destination_file_name
+      File.basename(migration_file, File.extname(migration_file)) +
+        ".#{plugin_name}" +
+        File.extname(migration_file)
+    end
+
+    def migration_file
+      @migration_file ||= begin
+        full_name = Dir.glob(File.join(migration_destination + 'db/migrate/*.rb')).grep(/\d+_#{migration_pattern}.rb$/).first
+        File.basename(full_name)
+      end
+    end
+
+    def plugin_source
+      # assuming a default folder layout for developer setup.
+      @plugin_source ||= options[:plugin_source] || "../#{plugin_name}/"
+    end
+
+    def plugin_name
+      @plugin_name ||= options[:plugin_name].underscore
```

Comment:
arguments usually use `--plugin-name` notation, the underscore should be replaced with dash. Other migration generator arguments do it at least.